js_modules/dagit/packages/core/src/ticks/EvaluateSchedule.tsx

```diff
@@ -0,0 +1,400 @@
+    viewport.width,
+  ]);
+
+  const buttons = React.useMemo(() => {
+    if (!shouldEvaluate) {
+      return (
+        <>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button
+            data-testid={testId('evaluate')}
+            intent="primary"
+            onClick={() => {
+              setShouldEvaluate(true);
+            }}
+          >
+            Evaluate
+          </Button>
+        </>
+      );
+    } else {
+      return <Button onClick={onClose}>Close</Button>;
+    }
+  }, [onClose, shouldEvaluate]);
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      style={{width: '70vw', display: 'flex'}}
+      title={
+        <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+          <Icon name="schedule" />
+          <span>{name}</span>
+        </Box>
+      }
+    >
+      <DialogBody>
+        <div style={{minHeight: '300px'}}>{content}</div>
+      </DialogBody>
+      {buttons ? <DialogFooter topBorder>{buttons}</DialogFooter> : null}
+    </Dialog>
+  );
+};
+
+export const GET_SCHEDULE_QUERY = gql`
+  query getSchedule(
+    $scheduleSelector: ScheduleSelector!
+    $startTimestamp: Float
+    $ticksAfter: Int
+    $ticksBefore: Int
+  ) {
+    scheduleOrError(scheduleSelector: $scheduleSelector) {
+      __typename
+      ... on PythonError {
+        message
+        stack
+      }
+      ... on Schedule {
+        name
+        potentialTickTimestamps(
+          startTimestamp: $startTimestamp
+          upperLimit: $ticksAfter
+          lowerLimit: $ticksBefore
+        )
+      }
+    }
+  }
+`;
+
+const EvaluateScheduleContent: React.FC<{
+  repoAddress: RepoAddress;
+  name: string;
+  timestamp: number;
+  jobName: string;
+}> = ({repoAddress, name, timestamp, jobName}) => {
+  const [userTimezone] = React.useContext(TimezoneContext);
+  const [scheduleDryRunMutation] = useMutation<
+    ScheduleDryRunMutation,
+    ScheduleDryRunMutationVariables
+  >(
+    SCHEDULE_DRY_RUN_MUTATION,
+    React.useMemo(() => {
+      const repositorySelector = repoAddressToSelector(repoAddress);
+      return {
+        variables: {
+          selectorData: {
+            ...repositorySelector,
+            scheduleName: name,
+          },
+          timestamp,
+        },
+      };
+    }, [name, repoAddress, timestamp]),
+  );
+  const [result, setResult] = React.useState<Awaited<
+    ReturnType<typeof scheduleDryRunMutation>
+  > | null>(null);
+  React.useEffect(() => {
+    scheduleDryRunMutation().then((result) => {
+      setResult(() => result);
+    });
+  }, [scheduleDryRunMutation]);
+
+  if (!result || !result.data) {
+    return (
+      <Box padding={32} flex={{justifyContent: 'center', alignItems: 'center'}}>
+        <Spinner purpose="page" />
+      </Box>
+    );
+  }
+
+  const evaluationResult =
+    result?.data?.scheduleDryRun.__typename === 'DryRunInstigationTick'
+      ? result?.data?.scheduleDryRun.evaluationResult
+      : null;
```

Comment:
This might not be in-scope, but this seems like a place where we'd have `evaluationResult` or a `PythonError`. Any chance there's more info than "unknown error" we could display here?

Reply:
the error is inside evaluationResult in this API. Not sure exactly why I think the underlying object is used in other APIs where there can be errors too